configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java

```diff
@@ -682,6 +682,77 @@ private boolean sessionLifeTimeElapsed(Instant created) {
+        int deleted = 0;
+        try {
+            for (Long sessionId : sessions) {
+                // Skip sessions newly added (we might have a session in the file system, but not in ZooKeeper,
+                // we will exclude these)
+                if (newSessions.contains(sessionId)) continue;
+
+                Session session = remoteSessionCache.get(sessionId);
+                if (session == null)
+                    session = new RemoteSession(tenantName, sessionId, createSessionZooKeeperClient(sessionId));
+
+                Instant createTime = session.getCreateTime();
+                Session.Status status = session.getStatus();
+                if (session.getStatus() == ACTIVATE && sessionIsActiveForApplication.test(session)) continue;
+
+                boolean hasExpired = hasExpired(createTime);
+                if (hasExpired) {
+                    log.log(Level.FINE, () -> "Remote session " + sessionId + " for " + tenantName + " has expired, deleting it");
+                    deleteRemoteSessionFromZooKeeper(session);
+                    deleted++;
+                }
+
+                log.log(Level.FINE, () -> "Candidate local session for deletion: " + sessionId +
+                        ", created: " + createTime + ", status " + status + ", can be deleted: " + canBeDeleted(sessionId, status) +
+                        ", hasExpired: " + hasExpired);
+                if (hasExpired && canBeDeleted(sessionId, status)) {
+                    log.log(Level.FINE, () -> "expired: " + hasExpired + ", can be deleted: " + canBeDeleted(sessionId, status));
+                    deleteLocalSession(sessionId);
+                } else if (createTime.plus(Duration.ofDays(1)).isBefore(clock.instant())) {
+                    LocalSession localSession;
+                    log.log(Level.FINE, () -> "more than 1 day old: " + sessionId);
+                    try {
+                        localSession = getSessionFromFile(sessionId);
+                    } catch (Exception e) {
+                        log.log(Level.FINE, () -> "could not get session from file: " + sessionId + ": " + e.getMessage());
+                        continue;
+                    }
+                    Optional<ApplicationId> applicationId = localSession.getOptionalApplicationId();
+                    if (applicationId.isEmpty()) continue;
+
+                    if ( ! sessionIsActiveForApplication.test(localSession)) {
+                        log.log(Level.FINE, () -> "Will delete inactive session " + sessionId + " created " +
+                                createTime + " for '" + applicationId + "'");
+                        deleteLocalSession(sessionId);
+                    }
+                }
+                if (deleted >= deleteMax)
+                    return;
+            }
+        } catch (Throwable e) { // Make sure to catch here, to avoid executor just dying in case of issues ...
```

Comment:
Should the try-catch be per session, to avoid one failing session blocking deletion of others? Alternatively, use a random or time-dependent subset of sessions—perhaps the first 0.05, but in reverse creation time order?

Reply:
Ah, they're in somewhat random order already, from being a hash set of longs ... 
and using reverse creation order potentially leaves older sessions hanging, of course. 
So some true randomness, then, or moving the try-catch down a step.